regex/Regex/Regex/Elab.lean

```diff
@@ -78,6 +78,14 @@ instance : ToExpr Regex where
     let maxTag := toExpr re.maxTag
     mkApp4 (mkConst ``Regex.mk) nfa wf maxTag (mkConst ``false)
 
+/--
+Macro for creating a regex at compile time.
+
+Parses a string literal into a `Regex` object during Lean's elaboration phase.
+This allows regex patterns to be checked at compile time rather than runtime.
+
+Example: `re! "a+b*c?"` creates a compiled regex matching that pattern
```

Comment:
Can you write that we can combine the macro with a raw string literal, e.g., `re! r"\d"`.